include/aie/Analysis/BufferDescriptorAnalysis.h

```diff
@@ -0,0 +1,90 @@
+//===- BufferDescriptorAnalysis.h -----------------------------------------===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2023 Xilinx Inc.
+//
+//===----------------------------------------------------------------------===//
+
```

Comment:
What does this do?  Is it really specific to your target, or is it more general?

Reply:
It's more to the target. Any suggestion on where this header file should live?